src/ports/postgres/modules/graph/hits.py_in

```diff
@@ -0,0 +1,439 @@
+        cur_join_clause = "{edge_temp_table}.{dest} = {cur}.{vertex_id}".format(**locals())
+        v1_join_clause = "{v1}.{vertex_id} = {edge_temp_table}.{src}".format(**locals())
+
+        authority_init_value = 1.0
+        hub_init_value = 1.0
+        plpy.execute("""
+                CREATE TEMP TABLE {cur} AS
+                SELECT {vertex_id}, {authority_init_value}::DOUBLE PRECISION AS authority,
+                {hub_init_value}::DOUBLE PRECISION AS hub
+                FROM {vertex_table}
+                {cur_distribution}
+            """.format(**locals()))
+
+        # The summary table contains the total number of iterations.
+        plpy.execute("""
+                CREATE TABLE {summary_table} (
+                    __iterations__ INTEGER
+                )
+            """.format(**locals()))
+
+        # Find all nodes in the graph that don't have any incoming edges and
+        # assign 0 as their authority and hub values.
+        no_incoming_node_authority_value = 0.0
+        no_incoming_node_hub_value = 0.0
+        plpy.execute("""
+                CREATE TABLE {nodes_with_no_incoming_edges} AS
+                SELECT DISTINCT({src}), {no_incoming_node_authority_value}::DOUBLE PRECISION AS authority,
+                {no_incoming_node_hub_value}::DOUBLE PRECISION AS hub
+                FROM {edge_temp_table}
+                EXCEPT
+                    (SELECT DISTINCT({dest}), {no_incoming_node_authority_value}::DOUBLE PRECISION AS authority,
+                    {no_incoming_node_hub_value}::DOUBLE PRECISION AS hub
+                    FROM {edge_temp_table})
+            """.format(**locals()))
+
+        unconverged_authority_num = 0
+        unconverged_hub_num = 0
+        iteration_num = 0
+        authority_norm = 0
+        hub_norm = 0
+
+        plpy.execute("""
+                    CREATE TABLE {message_unconv_authority} AS
+                    SELECT {cur}.{vertex_id}
+                    FROM {cur}
+                    LIMIT 0
+                    """.format(**locals()))
+        plpy.execute("""
+                    CREATE TABLE {message_unconv_hub} AS
+                    SELECT {cur}.{vertex_id}
+                    FROM {cur}
+                    LIMIT 0
+                    """.format(**locals()))
+
+        for iteration_num in range(max_iter):
+        ##########################################################################
+        # HITS scores for node 'A' at any given iteration 'i' is calculated as following:
+        # authority_i(A) = hub_i(B) + hub_i(C) + ...
+        # After calculating authority score for all nodes, a normalization will 
+        # be done for authority_i(A), authority_i(B), authority_i(C), ... 
+        # Then
+        # hub_i(A) = authority_i(B) + authority_i(C) + ... 
+        # After calculating hub score for all nodes, a normalization will 
+        # be done for hub_i(A), hub_i(B), hub_i(C), ...
+        # B, C are nodes that have edges that point to node A
```

Comment:
Can you please double check the algorithm? It looks like the order in which you perform updates and normalize must be different. I checked the source code in Wikipedia and it is similar to what you have done, but the original paper seems to be different. Look at page 9 in https://www.cs.cornell.edu/home/kleinber/auth.pdf.
```
auth_{i} <- USE hub_{i-1} and auth_{i-1}
hub_{i} <- USE hub_{i-1} and auth_{i}
auth_{i} <- normalize(auth_{i})
hub_{i} <- normalize(hub_{i})
```